contrib/scripts/print-downgrade-version.sh

```diff
@@ -1,15 +1,38 @@
 #!/usr/bin/env bash
 #
-# A utility script to print the branch name of the previous stable release.
+# A utility script to print the branch name of the previous stable or patch
+# release.
+
+set -o errexit
+set -o nounset
 
 SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
-VERSION="$(cat "$SCRIPT_DIR/../../VERSION")"
-if [[ $VERSION =~ ([0-9^]+)\.([0-9^]+)\..* ]] ; then
+VERSION=${VERSION-"$(cat "$SCRIPT_DIR/../../VERSION")"}
+if [[ $VERSION =~ ([0-9^]+)\.([0-9^]+)\.([0-9^]+).* ]] ; then
     major=${BASH_REMATCH[1]}
     minor=${BASH_REMATCH[2]}
-    ((minor--))
-    echo "v${major}.${minor}${BRANCH_SUFFIX:-}"
+    patch=${BASH_REMATCH[3]}
 else
-  echo "ERROR: failed to parse version '$VERSION'"
+  >&2 echo "ERROR: failed to parse version '$VERSION'"
   exit 1
 fi
+
+if [[ ${1-} == "patch" ]] ; then
+    # If user passed "patch" as first argument, print the latest patch version
+    case ${patch} in
+        0|90)
```

Comment:
(Not to address in this PR. Just trying to start that discussion.)

Should we try to test up/downgrades from/to release candidates as well? Would the last release candidate be the version prior to the `.0` here?

Reply:
> Should we try to test up/downgrades from/to release candidates as well?

I haven't worked with IPsec as much as you have, and I don't have the same hindsight about it, so... you tell me? I suppose that yes, we could add upgrades/downgrades with previews next, to test the latest changes on `main`.

> Would the last release candidate be the version prior to the `.0` here?

No, with the current scheme we always get `.0` on `main`. I don't think we have a way to find previews with the current script, we would probably need to download the Git history in the workflow and look at tags.

Reply:
Nothing IPsec specific here but yes, there could be IPsec changes in release candidates as well. Knowing about regressions before the .0 would probably be useful.

My main concern is that we'd grow the script more complex and an error in the script can cause some tests to be incorrectly skipped today.

Reply:
> My main concern is that we'd grow the script more complex and an error in the script can cause some tests to be incorrectly skipped today.

Sounds like a legitimate concern, we could probably come up with a different script for that case though, given that we won't be able to infer the release candidate commits from the `VERSION` file. We'd need some additional conditions in the GitHub workflow to run a different script for {up,down}grades with previews. Including a way to properly figure out we're on the `main` branch (we don't want to find and test the latest preview on stable branches).

I can take a look after this PR, if this is something we want.

Reply:
Let's postpone for now. There are higher priority tasks around CI and IPsec.